extensions/amp-base-carousel/1.0/base-carousel.type.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @externs */
+
+/** @const */
+var BaseCarouselDef = {};
+
+/**
+ * @typedef {{
+ *   arrowNext: (?PreactDef.VNode|undefined),
+ *   arrowPrev: (?PreactDef.VNode|undefined),
+ *   children: (!PreactDef.Renderable),
+ *   defaultSlide: (number|undefined),
+ *   loop: (boolean|undefined),
+ *   slide: (number|undefined),
+ *   onMount: (function():undefined|undefined),
```

Comment:
This shouldn't be a public property. IMHO either let's make it private or remove.

Reply:
I can make it private for now. For my learning, why don't we want publishers to be able to access a common `onMount` hook on all Preact components?

Reply:
I don't think I see this a lot. In theory, when I do `<MyComponent>` it's always mounted. So it'd seem like it'd be a rare need, at the very least.

Reply:
Also, I'm going to start replacing owner system with context props. So I think this will become unneeded soon anyway.

Reply:
In that case I will remove this `onMount` and the section where `amp-base-carousel` opts into the owners system to allow the AMP runtime to do this for now. I will also hold off on any `ChildLayoutManager` work for now.